File: app/src/main/java/com/OverSadBoy/samurairise/view/ItemAdapter.java
package com.OverSadBoy.samurairise.view;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.SwitchCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.OverSadBoy.samurairise.R;
import com.OverSadBoy.samurairise.model.database.DataBase;
import com.OverSadBoy.samurairise.model.database.Item;

import java.util.ArrayList;
import java.util.List;

public class ItemAdapter extends RecyclerView.Adapter<ItemAdapter.ViewHolder> {

    private List<Item> data;

    ItemAdapter(ArrayList<Item> alarms) {
        data = alarms;
        notifyDataSetChanged();
    }

    public void addData(Item item) {
        data.add(item);
        notifyDataSetChanged();
    }

    public int getAlarmsCount(){
        return data.size();
    }

    private void updateStatus(Item item, boolean status) {
        String str = "выключен";
        if (status)
            str = "включен";
        item.setStatus(str);
        notifyDataSetChanged();
    }

    public void deleteData(Item item) {
        data.remove(item.getId());
        notifyDataSetChanged();
    }


    @NonNull
    @Override
    public ItemAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ItemAdapter.ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_alarm, parent, false));
    }

    @Override
    public void onBindViewHolder(ItemAdapter.ViewHolder holder, int position) {
        holder.bind(data.get(position));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView timeAlarm;
        private final TextView settingsAlarm;
        private final SwitchCompat switchCompat;

        ViewHolder(View itemView) {
            super(itemView);
            timeAlarm = itemView.findViewById(R.id.timeAlarm);
            settingsAlarm = itemView.findViewById(R.id.settingsAlarm);
            switchCompat = itemView.findViewById(R.id.statusAlarm);
        }

        void bind(Item item) {
            timeAlarm.setText(item.getTime());
            settingsAlarm.setText(String.format("" +
                    "%s | %s", item.getRepeat(), item.getStatus()));
            switchCompat.setChecked(item.getStatus().equals("включен"));
            switchCompat.setOnCheckedChangeListener((compoundButton, b) -> updateStatus(item, switchCompat.isChecked()));

        }
    }
}

File: app/src/main/java/com/overSadBoy/samurairise/view/adapter/ItemAdapter.kt
package com.overSadBoy.samurairise.view.adapter

import android.util.SparseBooleanArray
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import android.widget.TextView
import androidx.appcompat.widget.SwitchCompat
import androidx.recyclerview.widget.RecyclerView
import androidx.test.runner.AndroidJUnit4
import com.overSadBoy.samurairise.R
import com.overSadBoy.samurairise.model.Alarm
import org.junit.runner.RunWith
import java.util.*

class ItemAdapter internal constructor(alarms: ArrayList<Alarm?>?) : RecyclerView.Adapter<ItemAdapter.ViewHolder?>() {
    private val data: MutableList<Alarm?>?
    private var listener: ItemsAdapterListener? = null
    fun setListener(listener: ItemsAdapterListener?) {
        this.listener = listener
    }

    fun addData(alarm: Alarm?) {
        data.add(alarm)
        notifyDataSetChanged()
    }

    fun getAlarmsCount(): Int {
        return data.size
    }

    private fun updateStatus(alarm: Alarm?, status: Boolean) {
        var str = "выключен"
        if (status) str = "включен"
        alarm.setStatus(str)
        notifyDataSetChanged()
    }

    fun deleteData(id: Int) {
        data.removeAt(id)
        notifyDataSetChanged()
    }

    fun switchStatus(alarm: Alarm?, status: Boolean) {
        if (status) data.get(alarm.getId()).setStatus("включен") else data.get(alarm.getId()).setStatus("выключен")
        notifyDataSetChanged()
    }

    private val selections: SparseBooleanArray? = SparseBooleanArray()
    private val selectionAlarm: ArrayList<Alarm?>? = ArrayList()
    fun toggleSelection(alarm: Alarm?, position: Int) {
        if (selections.get(position, false)) {
            selections.delete(position)
            selectionAlarm.remove(alarm)
        } else {
            selections.put(position, true)
            selectionAlarm.add(alarm)
        }
        notifyItemChanged(position)
    }

    fun clearSelections() {
        selections.clear()
        selectionAlarm.clear()
        notifyDataSetChanged()
    }

    fun getSelectedItems(): ArrayList<Alarm?>? {
        return selectionAlarm
    }

    fun getSelectedItemsPosition(): ArrayList<Int?>? {
        val items = ArrayList<Int?>(selections.size())
        for (i in 0 until selections.size()) {
            items.add(selections.keyAt(i))
        }
        return items
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_alarm, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
        holder.bind(data.get(position), position, listener, data.get(position).getStatus() == "включен", selections.get(position, false))
    }

    override fun getItemCount(): Int {
        return data.size
    }

    internal inner class ViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView) {
        private val timeAlarm: TextView?
        private val settingsAlarm: TextView?
        private val switchCompat: SwitchCompat?
        fun bind(alarm: Alarm?, position: Int, listener: ItemsAdapterListener?, status: Boolean, selected: Boolean) {
            itemView.setOnClickListener { v: View? ->
                listener?.onItemClick(alarm, position)
            }
            itemView.setOnLongClickListener { v: View? ->
                listener?.onItemLongClick(alarm, position)
                true
            }
            itemView.isActivated = selected
            timeAlarm.setText(alarm.getTime())
            settingsAlarm.setText(String.format("" +
                    "%s | %s", alarm.getRepeat(), alarm.getStatus()))
            switchCompat.setChecked(status)
            switchCompat.setOnCheckedChangeListener(CompoundButton.OnCheckedChangeListener { compoundButton: CompoundButton?, b: Boolean -> listener?.onSwitchClick(alarm, b, position) })
        }

        init {
            timeAlarm = itemView.findViewById<TextView?>(R.id.timeAlarm)
            settingsAlarm = itemView.findViewById<TextView?>(R.id.settingsAlarm)
            switchCompat = itemView.findViewById(R.id.statusAlarm)
        }
    }

    init {
        data = alarms
        notifyDataSetChanged()
    }
}